### ISSUE TITLE
[SUGGESTION] Add multiple song selection + actions

### ISSUE THREAD

**Issue by andiandi13** [vanilla-music/vanilla#792]
Vanilla Music Version : 1.0.60
- Android Version: 4.4.4
- Phone vendor / model: Galaxy S5
- Application was downloaded from :
  - [X] Play Store
  - [ ] F-Droid
  - [ ] Other (Nightly Build)
- Did this work in a previous version?
  - [ ] Yes
  - [X] No
  - [ ] NA / Don't know.

## Description

As you improved playlist management, I began to created playlists, but I "realized" it's long and hard to add multiple files.

Before that I added songs rarely so it wasn't that bad.

Would it be hard to add multiple selection function in the code (with checkboxes for example) and then actions, e.g **add to playlist**, **add to queue** etc.. ?

I know its SQL view but I'm not sure if that's a problem.

**Comment by adrian-bl** [comment]
I think that would be a duplicate of #90 ?

Adding support for this should now be simple as almost all views use the same view class (which already does support checkboxes)

**Comment by andiandi13** [comment]
> I think that would be a duplicate of #90 ?

Yes kinda, I'm sorry I thought no one asked this ever.

> Adding support for this should now be simple as almost all views use the same view class (which already does support checkboxes)


That's a good news ! It'll really improve the overall experience : we could delete, enqueue, add to playlist multiple tracks, what better ?

For now, I'm using AIMP to create m3u files but honesty, I'm fed up using 20 apps to manage my musics.

I look forward to see it implemented.

Thank you.

**Comment by Adonai** [comment]
@adrian-bl , you don't need checkboxes for that, better take a look at [action mode](https://developer.android.com/guide/topics/ui/menus#CAB).

**Comment by adrian-bl** [comment]
Yes, you are right.

But not sure how well that will work with the queuefragment

**Comment by thigg** [comment]
Sorry didnt find this issue. Closed the duplicate:

IIt would be nice if multiple items in Library view could be selected.
On such a selection actions such as

    Add all to playlist (for example only a few songs of an album)
    Edit tags of selection

could be performed.

My usecase is the second one. It is really annoying to edit the tags of an artist per song.

It would be possible to have a menu item which says enable multi select, which enables the feature and then provides an action based on the selection.
Actions must depend on what is selected (folder, artist,album,file,song).
When the action is performed multi select could be disabled again and on long press the current context menu would appear.

This would enable a more intuitive and faster handling of libraries.

**Comment by andiandi13** [comment]
Yes, I really thing it's the biggest lacking feature on Vanilla.

**Comment by Adonai** [comment]
I second this, actually. Once Vanilla has an ability to select multiple songs I can start working on supporting it in plugins

**Comment by PureIncompetence** [comment]
Is this a goal which is still aimed for, @adrian-bl ?
ATM for me this is the only major drawback this player has against its (proprietary) alternatives.

**Comment by xeruf** [comment]
Please close either as this is an agreed duplicate of #90 - maybe close the other since there has been more activity here.
